static/js/protein/chart.ts

```diff
@@ -432,6 +460,280 @@ export function drawProteinInteractionChart(
+  nodeData.push(centerDatum);
+
+  const linkData: InteractionLink[] = nodeData.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeID,
+      target: node.id,
+    };
+  });
+
+  const height = 400 - MARGIN.top - MARGIN.bottom;
+  const width = 700 - MARGIN.left - MARGIN.right;
+
+  const svg = d3
+    .select(`#${elementID}`)
+    .append("svg")
+    .attr("width", width + MARGIN.left + MARGIN.right)
+    .attr("height", height + MARGIN.top + MARGIN.bottom)
+    .attr("viewBox", `${-width / 2} ${-height / 2} ${width} ${height}`)
+    .append("g")
+    .attr("transform", `translate(${MARGIN.left - 150}, ${MARGIN.top - 25})`)
+    .attr("style", "max-width: 100%; height: auto; height: intrinsic");
+
+  const nodeIDs = nodeData.map((node) => node.id);
+  const nodeDepths = nodeData.map((node) => node.depth);
+  const nodeColors = d3.scaleOrdinal(nodeDepths, NODE_STYLE.circles.fillColors);
+
+  // force display layout
+  const forceNode = d3.forceManyBody();
+  const forceLink = d3
+    .forceLink(linkData)
+    .id(({ index }) => `${nodeIDs[index]}`);
+  forceLink.distance(LINK_STYLE.length);
+
+  const simulation = d3
+    .forceSimulation(nodeData)
+    .force("link", forceLink)
+    .force("charge", forceNode)
+    .force("center", d3.forceCenter())
+    .on("tick", ticked);
+
+  const links = svg // links first so nodes appear over links
+    .append("g")
+    .attr("stroke", LINK_STYLE.stroke.color)
+    .attr("stroke-opacity", LINK_STYLE.stroke.opacity)
+    .selectAll("line")
+    .data(linkData)
+    .join("line")
+    .attr(
+      "stroke-width",
+      (link) => LINK_STYLE.stroke.scoreWidthMultiplier * link.score
+    )
+    .attr("stroke-linecap", LINK_STYLE.stroke.linecap)
+    .on("mouseover", lightenThis)
+    .on("mouseleave", darkenThis);
+
+  const nodes = svg // container for circles and labels
+    .append("g")
+    .selectAll("g")
+    .data(nodeData)
+    .enter()
+    .append("g")
+    .call(drag(simulation))
+    .on("mouseover", lightenThis)
+    .on("mouseleave", darkenThis);
+
+  const nodeCircles = nodes
+    .append("circle")
+    .attr("stroke", NODE_STYLE.circles.stroke.color)
+    .attr("stroke-opacity", NODE_STYLE.circles.stroke.opacity)
+    .attr("stroke-width", NODE_STYLE.circles.stroke.width)
+    .attr("r", NODE_STYLE.circles.radius)
+    .attr("fill", (node) => nodeColors(node.depth));
+
+  const nodeLabels = nodes
+    .append("text")
+    .attr("fill", NODE_STYLE.labels.font.color)
+    .attr("text-anchor", "middle")
+    .attr("dominant-baseline", "middle")
+    .style(
+      "font",
+      `${NODE_STYLE.labels.font.size} ${NODE_STYLE.labels.font.name}`
+    )
+    .text(({ name }) => `${name}`);
+
+  // floating name labels we can switch in
+  // const labels = node
+  //   .append("text")
+  //   .attr("fill", "#555")
+  //   .attr("dx", 15)
+  //   .attr("dy", -15)
+  //   .text(({name}) => `${name}`)
+
+  function ticked(): void {
```

Comment:
this and `drag()` should be functions that are defined outside of the `drawProteinInteractionGraph` function